api/tests/test_room_archive.py

```diff
@@ -0,0 +1,85 @@
+from asyncio import CancelledError
+from typing import AsyncIterator, Callable, Iterator
+from uuid import uuid4
+
+import aiobotocore
+import pytest
+from aiobotocore.client import AioBaseClient
+
+from localstack.services import infra
+from pytest_lazyfixture import lazy_fixture
+
+from src.room_store.room_archive import RoomArchive
+from src.room_store.s3_room_archive import S3RoomArchive
+from src.util.async_util import async_collect
+from tests.static_fixtures import VALID_ACTION
+
+pytestmark = pytest.mark.asyncio
+
+
+@pytest.fixture(scope='session')
+def s3_url() -> Iterator[str]:
+    infra.start_infra(asynchronous=True, apis=['s3'])
```

Comment:
Can we specify a port here? In general hard-coding ports in tests is not a great idea because the tests fail if something else is using that port (for example, if we're running tests in parallel).

Reply:
I think this is more work than it's worth right now.